### PR TITLE
FIXConnection changes to support more Channel types (Solution 1)

### PR DESCRIPTION
**Pull request by mrossi975**
hi,
this solution relies on generics to allow FIXConnection to specify the 2 interfaces in Channel that it needs:

public class FIXConnection<T extends ReadableByteChannel & GatheringByteChannel> implements Closeable {

The drawbacks on this is that now FIXConnection is generic and that might not be of everyone's taste.

I will shortly post another PR with an alternative solution based on a new interface

### CODE REVIEW

File: libraries/core/src/test/java/com/paritytrading/philadelphia/FIXInitiatorTest.java [hunk 1]
```diff
@@ -408,18 +410,22 @@ void sendApplicationMessage() throws IOException {
         initiatorMessages(messages);
     }
```
Comment (mrossi975) [inline libraries/core/src/test/java/com/paritytrading/philadelphia/FIXInitiatorTest.java:412]: hi, this is the only part where the codebase relies on blocking behaviour.
Is it still wanted / needed?

File: libraries/core/src/main/java/com/paritytrading/philadelphia/FIXConnection.java [hunk 2]
```diff
@@ -23,18 +23,20 @@
...
 import java.nio.ByteBuffer;
+import java.nio.channels.GatheringByteChannel;
+import java.nio.channels.ReadableByteChannel;
 import java.nio.channels.SocketChannel;
 import org.joda.time.DateTimeZone;
 import org.joda.time.MutableDateTime;
 
 /**
  * A connection.
  */
```
Comment (mrossi975) [inline libraries/core/src/main/java/com/paritytrading/philadelphia/FIXConnection.java:34]: this is basically the root of the proposed solution

**Comment by jvirtanen** [comment]
Thank you, @mrossi975! My apologies, I've been busy with work recently. I will have a look at these later this week. 🔍

**Comment by jvirtanen** [comment]
Thank you for this contribution, @mrossi975! 🙌🏼  Support for this feature just got merged in #238.

**Pull request closed without merging**